File: Lab07/CO523_Lab07_E20449/Task02.c
// Task 02 - C (pass-by-reference using pointers)

#include <stdio.h>

void modify(int *x) {
    *x = *x + 10;
    printf("Inside function: %d\n", *x);
}

int main() {
    int a;
    printf("Enter an integer: ");
    scanf("%d", &a);

    printf("Before function: %d\n", a);

    modify(&a);

    printf("After function: %d\n", a);
    return 0;
}

File: Lab07/CO523_Lab07_E20449/Task02.py
# Task 02 - Python (pass-by-reference simulation using mutable list)

def modify_list(lst):
    lst[0] = lst[0] + 10
    print("Inside function:", lst)

numbers = [int(input("Enter an integer: "))]
print("Before function:", numbers)

modify_list(numbers)

print("After function:", numbers)